plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/state/MapboxOverviewState.kt

```diff
@@ -0,0 +1,121 @@
+ * will automatically trigger another transition.
+ */
+class MapboxOverviewState(
+  mapDelegateProvider: MapDelegateProvider,
+  initialOptions: OverviewViewportStateOptions
+) : OverviewViewportState {
+  private val cameraPlugin = mapDelegateProvider.mapPluginProviderDelegate.camera
+  private val cameraDelegate = mapDelegateProvider.mapCameraManagerDelegate
+  private val transitionFactory = MapboxViewportTransitionFactory(mapDelegateProvider)
+  private val dataSourceUpdateObservers = CopyOnWriteArrayList<ViewportDataSourceObserver>()
+  private var runningAnimation: AnimatorSet? = null
+
+  private var isOverviewStateRunning = false
+
+  override var options: OverviewViewportStateOptions = initialOptions
+    set(value) {
+      field = value
+      handleNewData()
+    }
+
+  private fun handleNewData() {
+    val cameraOptions = evaluateViewportData()
+    if (isOverviewStateRunning) {
+      updateFrame(cameraOptions, false)
+    }
+    dataSourceUpdateObservers.forEach {
+      if (!it.onNewData(cameraOptions)
+      ) {
+        dataSourceUpdateObservers.remove(it)
+      }
+    }
+  }
+
+  private fun evaluateViewportData(): CameraOptions {
+    return cameraDelegate.cameraForGeometry(
+      options.geometry,
+      options.padding,
+      options.bearing,
+      options.pitch
+    )
+  }
+
+  override fun observeDataSource(viewportDataSourceObserver: ViewportDataSourceObserver): Cancelable {
+    dataSourceUpdateObservers.add(viewportDataSourceObserver)
+    viewportDataSourceObserver.onNewData(evaluateViewportData())
+
+    return Cancelable {
+      dataSourceUpdateObservers.remove(viewportDataSourceObserver)
+    }
+  }
+
+  override fun startUpdatingCamera(): Cancelable {
+    isOverviewStateRunning = true
+    return Cancelable {
+      isOverviewStateRunning = false
+      cancelAnimation()
+    }
+  }
+
+  private fun cancelAnimation() {
+    runningAnimation?.apply {
+      cancel()
+      childAnimations.forEach {
+        cameraPlugin.unregisterAnimators(it as ValueAnimator)
+      }
+    }
+    runningAnimation = null
+  }
+
+  private fun startAnimation(
+    animatorSet: AnimatorSet,
+    instant: Boolean,
+  ) {
+    cancelAnimation()
+    animatorSet.childAnimations.forEach {
+      cameraPlugin.registerAnimators(it as ValueAnimator)
+    }
+    if (instant) {
+      animatorSet.duration = 0
+    }
+    animatorSet.start()
+    runningAnimation = animatorSet
+  }
+
+  private fun finishAnimation(animatorSet: AnimatorSet?) {
+    animatorSet?.childAnimations?.forEach {
+      cameraPlugin.unregisterAnimators(it as ValueAnimator)
+    }
+    if (runningAnimation == animatorSet) {
+      runningAnimation = null
+    }
+  }
+
+  private fun updateFrame(cameraOptions: CameraOptions, instant: Boolean = false) {
+    startAnimation(
+      transitionFactory.transitionLinear(cameraOptions, options.frameTransitionMaxDurationMs)
+        .apply { doOnEnd { finishAnimation(this) } },
```

Comment:
afair `doOnEnd` requires JDK 8, we do need it declared explicitly in gradle file. however I might be wrong about it all but just noting it here 😉

Reply:
We do have it declared in the module's gradle file 😄